### PR TITLE
Remove support for legacy auth tokens (without user_id).

### PR DESCRIPTION
**Pull request by isoos**
#2281

### CODE REVIEW

**Comment by jonasfj** [comment]
This deserves a changelog entry.

File: app/lib/account/backend.dart [hunk 1]
```diff
@@ -398,6 +379,11 @@ class GoogleOauth2AuthProvider extends AuthProvider {
         return null;
       }
 
+      if (info.userId == null) {
```
Comment (jonasfj) [inline app/lib/account/backend.dart]: Just uncomment the lines above which say:
```
          // TODO: Enable when pub client is requesting the correct scopes.
          // info.userId == null ||
          // info.userId.isEmpty ||
```

Comment (isoos) [reply app/lib/account/backend.dart]: good catch!

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,6 +3,10 @@ AppEngine version, listed here to ease deployment and troubleshooting.
 
 ## Next Release (replace with git tag when deployed)
 
+ * Removed support for legacy auth tokens (without `user_id`).
```
Comment (sigurdm) [inline CHANGELOG.md]: Can we add something like:
```
Users with legacy tokens will be requested to login again.
```
?

Comment (isoos) [reply CHANGELOG.md]: Sure, added.

**Pull request merged**